src/main/cpp/webp_codec.cpp

```diff
@@ -0,0 +1,104 @@
+#include "webp_codec.h"
+#include "webp/decode.h"
+#include "webp/encode.h"
+#include <cstdint>
+#include <stdexcept>
+
+/*
+ * WEBPEncoder
+ */
+
+libench::WEBPEncoder::WEBPEncoder() {
+  WebPMemoryWriterInit(&this->writer_);
+};
+
+libench::WEBPEncoder::~WEBPEncoder() {
+  WebPMemoryWriterClear(&this->writer_);
+};
+
+libench::CodestreamContext libench::WEBPEncoder::encodeRGB8(const ImageContext &image) {
+  return this->encode8(image);
+}
+
+libench::CodestreamContext libench::WEBPEncoder::encodeRGBA8(const ImageContext &image) {
+  return this->encode8(image);
+}
+
+libench::CodestreamContext libench::WEBPEncoder::encode8(const ImageContext &image) {
+  WebPMemoryWriterClear(&this->writer_);
+
+  WebPConfig config;
+  WebPPicture pic;
+  const int level = 6;  // 0 (faster) - 9 (slower)
```

Comment:
Note 6 is the default which gives a balance between throughput and density. I can also parameterize this if you wanted to focus on throughput using 3 or below.